sentry_sdk/integrations/starlette.py

```diff
@@ -516,12 +516,16 @@ async def raw_data(self):
     async def form(self):
         # type: (StarletteRequestExtractor) -> Any
         """
-        curl -X POST http://localhost:8000/upload/somethign -H "Content-Type: application/x-www-form-urlencoded" -d "username=kevin&password=welcome123"
-        curl -X POST http://localhost:8000/upload/somethign  -F username=Julian -F password=hello123
+        curl -X POST http://localhost:8000/upload/something -H "Content-Type: application/x-www-form-urlencoded" -d "username=kevin&password=welcome123"
+        curl -X POST http://localhost:8000/upload/something  -F username=Julian -F password=hello123
         """
         if multipart is None:
             return None
 
+        # Parse the body first to get it cached, as Starlette does not cache form() as it
+        # does with body() and json() https://github.com/encode/starlette/discussions/1933
+        await self.request.body()
```

Comment:
this is getting more and more hacky tbh.

I'd rather prefer we rewrite the whole `extract_request_info` thing to something that makes sense for starlette. The current structure was just copied from flask and might not make sense for another framework. The relevant goal is to extract as much helpful request metadata as possible, the actual logic depends on the framework and what it supports.

Reply:
I know this looks hacky. But for our integration to get the form data from the request calling .body() and then .form() is the only way to not break our users Starlette projects.

Reply:
This reading of the `body()` is still there, but I have now refactored `extract_request_info` to make it clearer what is happening.

Reply:
I guess this is now better understandable and enough "non hacky". What do you think @sl0thentr0py ?